### PR TITLE
[jit] Move TorchScript language reference to its own page

### PR DESCRIPTION
**Pull request by driazati**
Stacked PRs
 * #31146 - [jit] Cleanup after moving language reference
 * **#31138 - [jit] Move TorchScript language reference to its own page**

Preview: https://driazati.github.io/pytorch_doc_previews/jit.html#torchscript-language

Differential Revision: [D19167375](https://our.internmc.facebook.com/intern/diff/19167375/)

### CODE REVIEW

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 7dac11d:
* **1/1** broken upstream at merge base 679b20b (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=679b20b1e4b79dd3d62d8a8cb625efeb5a51fe29&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=7dac11d196d526084fefe02d0558e9ee479e12dd).

### 1 failure *not* recognized by patterns:
| Job | Step |
| --- | --- |
| CircleCI <sup>pytorch_windows_build</sup> | <sup>Build</sup> |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

**Comment by eellison** [comment]
I actually misinterpreted this when you first brought this up. I think we should have the language reference as the main landing page, and from there link to all the other pages. @suo thoughts ?

**Comment by driazati** [comment]
I think most people coming to the TorchScript docs will be looking for higher level info on stuff like: APIs (`script`, `trace`, etc), examples, and best practices. The language reference seems too low level for users to get a good idea of what TorchScript is since it has too many details

**Comment by suo** [comment]
Yeah I was thinking about our doc structure a bit, some general ideas:
- We should probably have our own "TorchScript" section on the sidebar, distinct from the `torch.jit` Python API.
- `torch.jit` Python API section should just be reference documentation for the public Python API.

Under the TorchScript section could be:
- Language reference.
- We should include more expository documentation (like the nn.Module -> ScriptModule quip I wrote internally) as well.
- TorchScript C++ API documentation and reference.

Anyway, I'm fine with moving the language reference to its own page in this PR, and we can work on the rest of the stuff spearately.

**Review by suo — APPROVED** [review]
Accepting as this seems like a fine incremental change.

**Comment by VitalyFedyunin** [comment]
Breaks master tests see https://circleci.com/gh/pytorch/pytorch/3980538?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by VitalyFedyunin** [comment]
Im reverting it now cc @driazati

File: docs/source/jit.rst [hunk 1]
```diff
@@ -1413,22 +538,6 @@ Built-in Functions and Modules
...
 
-PyTorch Functions and Modules
------------------------------
-
-TorchScript supports a subset of the tensor and neural network
-functions that PyTorch provides. Most methods on Tensor as well as functions in
-the ``torch`` namespace, all functions in ``torch.nn.functional`` and
-most modules from ``torch.nn`` are supported in TorchScript.
-
-See :ref:`jit_unsupported` for a list of unsupported PyTorch functions and modules.
```
Comment (eellison) [inline docs/source/jit.rst]: You overwrote my changes :'(

**Pull request closed without merging**